src/main/scala/com.socrata.geoexport/http/ExportService.scala

```diff
@@ -83,19 +87,20 @@ class ExportService(sodaClient: UnmanagedCuratedServiceClient) extends SimpleRes
   }
 
   private def getUpstreamLayers(req: HttpRequest,
-                                fxfs: Seq[String]): Either[HttpResponse, Seq[Response with Closeable]] = {
-    fxfs.map { fbf =>
-      val soql = req.queryParameter("query").getOrElse(s"select * limit ${Int.MaxValue}")
-      val copy = req.queryParameter("copy").getOrElse("published")
-      val context = req.queryParameter("context").getOrElse("{}")
+                                layers: Seq[LayerWithQuery]): Either[HttpResponse, Seq[Response with Closeable]] = {
+    val defaultSoql = s"select * limit ${Int.MaxValue}"
+    val defaultCopy = "published"
+    val defaultContext = "{}"
+
+    layers.map { layer =>
       val reqBuilder = { base: RequestBuilder =>
         base
-          .path(Seq("resource", s"${resourceNameify(fbf)}.soqlpack"))
-          .addParameter(("$query", soql))
-          .addParameter(("$$copy", copy))
-          .addParameter(("$$context", context))
+          .path(Seq("resource", s"${resourceNameify(layer.uid)}.soqlpack"))
+          .addParameter(("$query", layer.query.getOrElse(defaultSoql)))
+          .addParameter(("$$copy", layer.copy.getOrElse(defaultCopy)))
+          .addParameter(("$$context", layer.context.getOrElse(defaultContext)))
           .addHeader(ReqIdHeader -> req.requestId)
-          .addHeader("X-Socrata-Lens-Uid" -> fbf)
+          .addHeader("X-Socrata-Lens-Uid" -> layer.uid)
```

Comment:
Woah. First time seeing this header. I realize you haven't introduced a change here, but can you give me a little background on this?

Reply:
gosh. I was going to say, no idea, but I was poking in git commit history and it looks like I did add this. I have no explanation other than I probably saw it was being sent by core to geo-export and soda-fountain, and decided to pass it along.

Looks like it was at one point related to views/rows loaded metrics?